cmd/bee/cmd/db.go

```diff
@@ -139,25 +157,55 @@ func dbExportCmd(cmd *cobra.Command) {
 				defer f.Close()
 				out = f
 			}
-			c, err := storer.Export(out)
+
+			tw := tar.NewWriter(out)
+			var counter int64
+			err = db.ReserveIterateChunks(func(chunk swarm.Chunk) (stop bool, err error) {
+				logger.Info("exporting chunk", "address", chunk.Address().String())
+				b, err := MarshalChunkToBinary(chunk)
+				if err != nil {
+					return true, fmt.Errorf("error marshaling chunk: %w", err)
+				}
+				hdr := &tar.Header{
+					Name: chunk.Address().String(),
+					Size: int64(len(b)),
```

Comment:
You can use this size to unmarshal. I dont think we need to marshal the size separately.

Reply:
We need to marshall the size of the data part separeatly since it's variable. `size = dataLength + data + stamp`
We can get rid of  `dataLength` if we reverse the marshalling order and write the fixed length part (stamp) first.
But it's still a good practice to write the length of something that's not fixed.

Reply:
So if we have 4Million chunks, each entry would add 4bytes of length. So this is 16MB additional weight in the tar file which could have been easily saved.

Reply:
I'll update this in #4107.